### DISCUSSION TITLE
Response code lens missing

### DISCUSSION THREAD

**Discussion by asherber** [AnWeber/vscode-httpyac#160] [Q&A]
When I send a request, the response window used to show the elapsed time, content type, and content length in a code lens at the top. I don't think I've changed any settings in a while, but I'm not seeing that now. I do have the default:

```json
"httpyac.responseViewHeader": [
  "timings.total",
  "content-type",
  "content-length"
]
```
I don't see a setting to turn this code lens on and off. Am I missing something?

**Comment by AnWeber** [comment]
No for this CodeLens display I have not yet provided a config setting to turn it off. Also, the display still works for me even with the latest VSCode version (v1.73) and using your settings. Setting `editor.codeLens` allows all CodeLens to be deactivated. Can this possibly be the reason, or are other CodeLens still visible?

**Reply by asherber** [reply]
I can see other CodeLenses, like the ones for httpyac in my main code window, but I don't see the one in the response window. I tried looking in Dev Tools for errors but didn't see any.

Here are all my non-default httpyac settings:

```json
"httpyac.environmentPickMany": false,
"httpyac.environmentSelectedOnStart": [
    "dev"
],
"httpyac.environmentShowStatusBarItem": true,
"httpyac.responseViewExtensionRecognition": "mimetype",
"httpyac.responseViewLanguageMap": {
    "application/json": "json",
    "application/soap+xml": "xml",
    "text/html": "html",
    "text/xml": "xml"
},
"httpyac.testControllerEnabled": false,
"httpyac.useDecorationProvider": false,
```

**Reply by asherber** [reply]
I removed all of my non-default httpyac settings, and I still don't see the response CodeLens. Interestingly, if I send a request that receives an empty response (valid HTTP status, no content), I do see a different CodeLens:

![Screenshot_1](https://user-images.githubusercontent.com/5248041/200173899-d76720be-c6f8-4966-b9ff-7ebf46bf0dc5.png)

That CodeLens appears whether or not I have my custom settings in place. 

Could this be a conflict with a different extension? I can try disabling and selectively enabling to try to find the issue.

**Reply by AnWeber** [reply]
No, the reason is that I only display the response body by default. But for empty response I use the URL and the headers. For the formatting I set the language Http, which in turn causes the parsing as a normal HTTP file. 
A possible reason is/was possibly that I display the file first and only then insert it in the cache. But to display the CodeLens, I try to find the displayed file in the cache again. I have changed [the order](https://github.com/AnWeber/vscode-httpyac/commit/13ea274b2f264b99e7a12df0f5a5d5ce97d9a57d#diff-f85b08a09620e833d4504ff0a9ed8e4ed3188477139ddd28ee2f20ca4bdb2fe0).  If that doesn't fix the problem, I added some more log output and also made the ResponseStore visible in the GUI. The visible response filename should match uri object `responseCache[?].documentUri`.

<img width="206" alt="image" src="https://user-images.githubusercontent.com/13434739/200192826-fcfe6baa-0acb-45c4-809d-9fd933ce424b.png">

**Reply by asherber** [reply]
Did you release a new version yesterday? I've got v5.8.0, and I'm still not seeing the response CodeLens.

**Reply by AnWeber** [reply]
Yes, I have provided a new version. Here it would be interesting if the currently displayed file is also correctly included in the `responseStore.responseCache`. The documentUri must be filled and inside a folder `_httpyac_` (e.g. `\\wsl.localhost\Ubuntu\home\andreas\.vscode-server\data\User\globalStorage\anweber.vscode-httpyac\_httpyac_`). If this is not the case, it would be interesting to know where the file is located.

<img width="824" alt="image" src="https://user-images.githubusercontent.com/13434739/200653120-6b026b00-3894-4caf-bbaf-a8b58c2d79b7.png">

> I noticed that in the history view maybe the `documentUri` is not always displayed. Maybe just fire a second request, so that a refresh takes place.

**Comment by asherber** [comment]
Sorry for the late reply; I was out of town for a while.

Using v5.8.1 I am still not seeing the CodeLens. The response document itself is located in `C:\Users\ASherber\AppData\Roaming\Code\User\globalStorage\anweber.vscode-httpyac\_httpyac_`, but the `documentUri` shows the same path prefixed by `vscode-userdata:` -- not sure if this is significant. `

**Reply by AnWeber** [reply]
The different scheme is the reason for the problem. Why VSCode uses vscode-userdata here is a mystery. This scheme is actually used for the settings. I'll look for ideas to solve the problem.

**Reply by AnWeber** [reply]
And I can reproduce it now, too. When using the CLI option `--user-data-dir`, VSCode seems to open the responses with scheme `vscode-userdata`.

**Reply by AnWeber** [reply]
@asherber Now I should have fixed the error. I have released v5.8.2 for it

**Reply by asherber** [reply]
Yes! It's working for me now. Thanks very much for the quick fix!

**Comment by beholdr** [comment]
I have 6.11.0 version and response lens is missing too.
Instead it shows send/env lens. I have exchange view for response. This is how it looks:

<img width="1169" alt="image" src="https://github.com/AnWeber/vscode-httpyac/assets/741973/457ae4aa-34d0-4ccb-a3db-e266e189a968">

Looks like it sees response as .http file and apply wrong lens for it. Do you know how I could fix it?

**Reply by AnWeber** [reply]
It could be that I got a bug in there due to changes in the last version. But it could also be a behavior when comparing vscode.Uri. I will have a look at it as soon as I am fitter again.
And maybe you could use Output Channel to look at headers.

**Reply by beholdr** [reply]
Thanks, mainly I need info about timings, which is presented in the history too.

**Reply by AnWeber** [reply]
A change of storage location was the cause. I have released a new version 6.11.1

**Reply by beholdr** [reply]
Thanks for quick fix!
